pkg/ir/ir_generator.go

```diff
@@ -533,47 +531,17 @@ func parseDefaultNumericColumn(el *parser.Expr) (*NumericColumn, error) {
 }
 
 func parseNumericColumn(el *parser.ExprList) (*NumericColumn, error) {
-	help := "NUMERIC([DENSE()|SPARSE()|col_name][, SHAPE])"
-
-	// 'shape' is optional in NUMERIC, so len(*el) may be 2 or 3
-	if len(*el) != 2 && len(*el) != 3 {
-		return nil, fmt.Errorf("bad NUMERIC expression format: %s, should be like: %s", *el, help)
-	}
-
-	// 1. NUMERIC(DENSE()/SPARSE()) phrases
-	if (*el)[1].Type == 0 {
-		fieldDesc, err := parseFieldDesc(&(*el)[1].Sexp)
-		if err != nil {
-			return nil, err
-		}
-		return &NumericColumn{FieldDesc: fieldDesc}, nil
-	}
-	// 1. NUMERIC(col_name, ...) phrases
-	key, err := expression2string((*el)[1])
+	fieldDesc, err := parseFieldDesc(el)
 	if err != nil {
-		return nil, fmt.Errorf("bad NUMERIC key: %s, err: %s, should be like: %s", (*el)[1], err, help)
-	}
-
-	shape := []int{1}
-	if len(*el) == 3 {
-		shape, err = parseShape((*el)[2])
-		if err != nil {
-			return nil, err
-		}
+		return nil, err
 	}
-
 	return &NumericColumn{
-		FieldDesc: &FieldDesc{
-			Name:     key,
-			DType:    Float, // default use float dtype if no DENSE()/SPARSE() provided
-			Shape:    shape,
-			IsSparse: false,
-		},
+		FieldDesc: fieldDesc,
 	}, nil
 }
 
 func parseBucketColumn(el *parser.ExprList) (*BucketColumn, error) {
-	help := "BUCKET([NUMERIC(...)|col_name], BOUNDARIES)"
+	help := "BUCKET([DENSE(...)|col_name], BOUNDARIES)"
```

Comment:
Can't we specify `SPARSE`  here?

Reply:
Yes. The `tf.feature_column.bucketized_column` does not support sparse tensor.